src/components/menu/menu-item/menu-item.component.js

```diff
@@ -38,26 +40,58 @@ const MenuItem = ({
   onSubmenuClose,
   overrideColor,
   rel,
-  isFocused,
   ...rest
 }) => {
-  const menuContext = useContext(MenuContext);
+  const {
+    inFullscreenView,
+    registerItem,
+    unregisterItem,
+    focusId,
+    menuType,
+  } = useContext(MenuContext);
+  const menuItemId = useRef(guid());
   const submenuContext = useContext(SubmenuContext);
+  const {
+    registerItem: registerSubmenuItem,
+    unregisterItem: unregisterSubmenuItem,
+    submenuFocusId,
+    updateFocusId: updateSubmenuFocusId,
+    handleKeyDown: handleSubmenuKeyDown,
+  } = submenuContext;
   const ref = useRef(null);
-  const focusFromMenu = isFocused;
-  const focusFromSubmenu = submenuContext.isFocused;
-  const isChildSearch = useRef(false);
+  const focusFromMenu = focusId === menuItemId.current;
+  const focusFromSubmenu = submenuFocusId
+    ? submenuFocusId === menuItemId.current
+    : undefined;
   const childRef = useRef();
-  const { inFullscreenView } = menuContext;
-  const childrenItems = React.Children.map(children, (child) => {
```

Comment:
It could also be the removal of this that has caused the bug I pointed out previously too.